docs/libraries/typespec-client-generator-core/reference/decorators.md

```diff
@@ -450,6 +450,47 @@ Create a ClientGenerator.Core operation group out of a namespace or interface
 interface MyInterface {}
 ```
 
+### `@override` {#@Azure.ClientGenerator.Core.override}
+
+Override the default client method generated by TCGC from your service definition
+
+```typespec
+@Azure.ClientGenerator.Core.override(override: Operation, scope?: valueof string)
+```
+
+#### Target
+
+: The original service definition
+`Operation`
+
+#### Parameters
+
+| Name     | Type             | Description                                                                                                   |
+| -------- | ---------------- | ------------------------------------------------------------------------------------------------------------- |
+| override | `Operation`      | : The override method definition that specifies the exact client method you want                              |
+| scope    | `valueof string` | The language scope you want this decorator to apply to. If not specified, will apply to all language emitters |
+
+#### Examples
+
+```typespec
+// main.tsp
+namespace MyService;
+
+model Params {
+ foo: string;
+ bar: string;
+}
+op myOperation(...Params): void; // by default, we generate the method signature as `op myOperation(foo: string, bar: string)`;
+
+// client.tsp
+namespace MyCustomizations;
+
+op myOperationCustomization(@bodyRoot params: Params): void;
+
+@@override(MyService.myOperation, MyCustomizations.myOperationCustomization);
```

Comment:
So, the name of the operation in client.tsp doesn't matter if it's used as an override? Codegen will use the operation name defined in main.tsp?

Reply:
yep the name in client.tsp doesn't matter, it will just use the service name